src/plugin.ts

```diff
@@ -26,18 +26,18 @@ import { bindPluginMethods, deprecateRedirectUri } from './utils';
 /**
  * @ignore
  */
-export const client: Ref<Auth0VueClient> = ref(null);
+export const client: Ref<Auth0VueClient | null> = ref(null);
```

Comment:
Suggestion: since you are assuming `Auth0VueClient` is not null/undefined in a bunch of places (with `foo as Auth0VueClient`)

You could make the assumption in one place, then you wouldn't have to cast it everywhere else 

```suggestion
export const client: Ref<Auth0VueClient> = ref(null) as unknown as Ref<Auth0VueClient>;
```

Reply:
I wonder if that would cause issues, as the client could still contain null, if they try and read its value before our plugin's `install` method was called.

Reply:
Does the end user ever reference the client directly? Because if they do changing it to `Auth0VueClient | null` would create TS errors, unless they put `if (client)` guards everywhere (or cast `as Auth0VueClient` themselves)

Reply:
> All the places we have as Auth0VueClient are supposed to run after the plugin is to be installed, and the client expected to be instantiated (else the plugin was not registered and can not function)

Also, are their any places you check if the plugin has been installed before referencing the client?

Reply:
> Does the end user ever reference the client directly?

Yeah, and they might try and reference it before the plugin ran, so it could be null. I do understand it might break compilation, but it actualy rectifies previously incorrect behavior I think.

I would prefer have it types as not-nullable, but I guess that could cause some issues as it's actually lying.

> Also, are their any places you check if the plugin has been installed before referencing the client?

No, but it should never reach the plugin methods that access the client, instead it should return undefined/null here: https://github.com/auth0/auth0-vue/blob/fix/234/src/index.ts#L41

Which isn't correctly typed as such as users are not supposed to ever call it in a Vue App that doesn't use our plugin, so we type it as non-nullable (even tho technically it could be null when misconfigured).
For the exposed client, that's a global variable which lives more outside of the vue bits, and is actualy used to be able to access the plugin from outside of the Vue components. As we have less control over the actual life-cycle there, I thought it makes sense to mark that as nullable.

Happy to reconsider tho, just wanted to point out the reasoning a bit better.

Reply:
Let me look into this again. I dont think `client` is exposed to our users.